lib/shoulda/matchers/rails_shim.rb

```diff
@@ -62,7 +62,7 @@ def generate_validation_message(
         def serialized_attributes_for(model)
           attribute_types_for(model).
             inject({}) do |hash, (attribute_name, attribute_type)|
-              if attribute_type.is_a?(::ActiveRecord::Type::Serialized)
+              if Object.const_defined?('ActiveRecord::Type::Serialized') && attribute_type.is_a?(::ActiveRecord::Type::Serialized)
```

Comment:
What about moving the `const_defined?` check to the beginning of the method? This way we avoid making this check for every attribute of the model and fix the Rubocop line length issue.

Something along those lines:
```ruby
type_serialized_defined = Object.const_defined?('ActiveRecord::Type::Serialized')
attribute_types_for(model).
            inject({}) do |hash, (attribute_name, attribute_type)|
              if type_serialized_defined && attribute_type.is_a?(::ActiveRecord::Type::Serialized)
```

Reply:
Thank you. I have changed the code accordingly.

If we want to support any serialized type we may change the code to the following:

```
          attribute_types_for(model).
            inject({}) do |hash, (attribute_name, attribute_type)|
              if attribute_type.respond_to?(:coder)
```
`ActiveRecord::Type::Serialized` is the only type that responds to `coder`.